quickwit/quickwit-search/src/search_job_placer.rs

```diff
@@ -177,13 +178,30 @@ impl SearchJobPlacer {
         let mut job_assignments: HashMap<SocketAddr, (SearchServiceClient, Vec<J>)> =
             HashMap::with_capacity(num_nodes);
 
+        let total_load: usize = jobs.iter().map(|job| job.cost()).sum();
+
+        // allow arround 1% disparity. Round up so we never end up in a case where
+        // target_load * num_nodes < total_load
+        // some of our tests needs 2 splits to be put on 2 different searchers. It makes sens for
+        // these tests to keep doing so (testing root merge). Either we can make the allowed
+        // difference stricter, find the right split names ("split6" instead of "split2" works).
+        // or modify mock_split_meta() so that not all splits have the same job cost
+        // for now i went with the mock_split_meta() changes.
+        const ALLOWED_DIFFERENCE: usize = 101;
+        let target_load = (total_load * ALLOWED_DIFFERENCE).div_ceil(num_nodes * 100);
```

Comment:
note this is not sufficient to make sure you can always pick a node.

I think the following works:

(total_load * ALLOWED_DIFFERENCE).div_ceil(num_nodes * 100).max(max_job_load + total_load.ceil_div(num_nodes));

Reply:
We know $J <= J_{max}$ by definition, so this implies.

$\forall n < N, L_n > {T-{J_{max}}} \over {N}$

Reply:
We know that $\sum L_n <= T - {J_max}$  (otherwise the algorithm would have terminated).
Therefore this in turn imply,

$T - J_{max} > T - J_{max}$, which is a contradiction.

Reply:
Do we need to use necessarily this bound? No not really, but we are certain this is "good enough" which is a comfortable guarantee. 

Is the current solution good enough? Actually no it is not.
Think about the case where we have K large splits with the same node affinity, and S searchers, with K << S.

Reply:
after discussion a bit, the current implementation actually seems to work, because it checks `current_load < target_load`, and not `current_load + job_weight < target_load`